sdk/test-utils/recorder/src/utils.ts

```diff
@@ -106,13 +107,61 @@ export function delay(milliseconds: number): Promise<void> | null {
  */
 export function parseUrl(url: string): any {
   const [cleanUrl, ...queryParts] = url.split(/[?&]/);
-  const query = queryParts.reduce((query: { [key:string]: any }, part) => {	
+  const query = queryParts.reduce((query: { [key: string]: any }, part) => {
     const [name, value] = part.split(/=/);
-    query[name] = decodeURIComponent(value.replace(/\+/g, ' '));
+    query[name] = decodeURIComponent(value.replace(/\+/g, " "));
     return query;
   }, {});
   return {
     url: cleanUrl,
     query
+  };
+}
+
+/**
+ * ONLY WORKS IN THE NODE.JS ENVIRONMENT
+ *
+ * Meant to be called during the playback for the node tests.
+ * 1. Takes the test filePath as argument.
+ * 2. Looks for the `recordings` folder in its hierarchical path.
+ * 3. Returns the full path of the `recordings` folder
+ *
+ * While running the tests, `filePath` can vary depending on location of the test files, examples below
+ *
+ * 1. If roll-up generated bundle files are being leveraged to run the tests
+ *    filePath = `<base path>\azure-sdk-for-js\sdk\storage\storage-blob\dist-test\index.node.js`
+ * 2. If ts complied dist-esm files are being used to run the tests
+ *    filePath = `<base path>\azure-sdk-for-js\sdk\storage\storage-blob\dist-esm\test\utils.spec.js`
+ *    filePath = `<base path>\azure-sdk-for-js\sdk\storage\storage-blob\dist-esm\test\node\utils.spec.js`
+ * 3. If `.spec.ts` test files are being used directly
+ *    filePath = `<base path>\azure-sdk-for-js\sdk\storage\storage-blob\test\utils.spec.ts`
+ *    filePath = `<base path>\azure-sdk-for-js\sdk\storage\storage-blob\test\node\utils.spec.ts`
+ * In the above example, no matter where the test files are,
+ *    the recordings are located at `<base path>\azure-sdk-for-js\sdk\storage\storage-blob\recordings\`.
+ * In order to playback the tests, exact location of the recordings is to be found,
+ *    this is done by checking the parent(s) folders until the `recordings` folder is found.
+ *
+ * @export
+ * @param {string} filePath
+ * @returns {string} location of the `recordings` folder
+ */
+export function findRecordingsFolderPath(filePath: string): string {
+  let path = require("path");
+
+  // Stripping away the file name
+  let recordingsFolderPath = path.resolve(filePath, "..");
+  try {
+    // While loop to find the `recordings` folder
+    while (!fs.existsSync(path.resolve(recordingsFolderPath, "recordings/"))) {
```

Comment:
if this somehow ended up outside the package directory, is it worth checking if we hit the root dir or not?

Reply:
```
      if (fs.existsSync(path.resolve(recordingsFolderPath, "package.json"))) {
        // package.json of the SDK is found but not the `recordings` folder
        // which is supposed to be present at the same level as package.json
        throw new Error(`'recordings' folder is not found at ${recordingsFolderPath}`);
      }
```
This looks for the `package.json` file and throws an error if that is found before the `recordings` folder.
In case, even if the `package.json` in the `sdk\service\service-sdk` folder is somehow mysteriously missed, we'll definitely capture the root's `package.json`. 
So, fine?

Reply:
I think it's not fine, but for a different reason - a missing package.json anywhere will result in an infinite loop because, surprise surprise, `path.resolve('/', '..')` is just `'/'` again. I think it's good to put a base case in here - if we're looking at the same path again, we've reached the root directory and its time to error out.

Reply:
Sounds good, I'll add this base case.

Reply:
Added